src/main/groovy/net/wasdev/wlp/gradle/plugins/tasks/InstallAppsTask.groovy

```diff
@@ -142,4 +142,154 @@ class InstallAppsTask extends AbstractServerTask {
+        }
+        LooseWarApplication looseWar = new LooseWarApplication(project, config)
+        looseWar.addSourceDir()
+        looseWar.addOutputDir(looseWar.getDocumentRoot(), project, "/WEB-INF/classes");
+
+        //retrieves dependent library jar files
+        addWarEmbeddedLib(looseWar.getDocumentRoot(), looseWar);
+
+        //add Manifest file
+        File manifestFile = new File(project.sourceSets.main.getOutput().getResourcesDir().getParentFile().getAbsolutePath() + "/META-INF/MANIFEST.MF")
+        looseWar.addManifestFile(manifestFile, "gradle-war-plugin")
+    }
+
+    private void addWarEmbeddedLib(Element parent, LooseApplication looseApp) throws Exception {
+      ArrayList<File> deps = new ArrayList<File>();
+      project.configurations.runtime.each { deps.add(it)}
+      project.configurations.providedRuntime.each { deps.remove(it) }
+      File parentProjectDir = new File(project.getRootProject().rootDir.getAbsolutePath())
+      for (File dep: deps) {
+        String projectPath = getProjectPath(parentProjectDir, dep)
+        if(!projectPath.isEmpty() && project.getRootProject().findProject(projectPath) != null){
+            Element archive = looseApp.addArchive(parent, "/WEB-INF/lib/"+ dep.getName());
+            looseApp.addOutputDirectory(archive, project.getRootProject().findProject(projectPath), "/");
+            looseApp.addManifestFile(archive, project.getRootProject().findProject(projectPath), "gradle-jar-plugin");
+        } else{
+            looseApp.getConfig().addFile(parent, dep.getAbsolutePath() , "/WEB-INF/lib/" + dep.getName());
+        }
+      }
+    }
+
+    private String getProjectPath(File parentProjectDir, File dep){
+      String projectJunk = dep.getAbsolutePath().replace(parentProjectDir.getAbsolutePath()+"/","")
+      String projectPath = dep.getAbsolutePath().replace(projectJunk,"")
+      Pattern pattern = Pattern.compile("/build/.*")
+      Matcher matcher = pattern.matcher(projectJunk)
+      projectPath = matcher.replaceAll("")
+      return projectPath;
+    }
+
+    private boolean containsJavaSource(){
+      Set<File> srcDirs = project.sourceSets.allJava.getSrcDirs();
+      for(srcDir in srcDirs){
+        File javaSourceDir = new File (srcDir)
+        if(javaSourceDir.exists() && javaSourceDir.isDirectory() && containsJavaSource(javaSourceDir)){
+          return true;
+        }
+      }
+      return false;
+    }
+
+    private boolean containsJavaSource(File f){
+      File[] files = f.listFiles()
+      for (file in Files) {
+            if (file.isFile() && file.getName().toLowerCase().endsWith(".java")) {
+                return true;
+            } else if (file.isDirectory()) {
+                return containsJavaSource(file);
+            }
+        }
+        return false;
+    }
+
+    private boolean isSupportedType(){
+      switch (getPackagingType()) {
+        case "ear":
+        case "war":
+        case "eba":
+        case "esa":
```

Comment:
we won't get to eba or esa at this time. remove.  
But I am thinking this method will need to process a task.  war, ear. etc.